### ISSUE TITLE
[a11y] Missing and incorrect heading semantics

### ISSUE THREAD

**Issue by jcsteh** [mozilla/addons-frontend#8868]
On add-on pages, there are several headers (Screenshots, About this extension, Other users with this extension also installed, Rate your experience, Permissions, More information, Add to collection, Release notes for {version}) which are not exposed as headings for accessibility. Given the amount of information on these pages, this makes it very difficult for a screen reader user to jump to the information they're interested in. For example, a user often wants to quickly read "About this extension", but right now, a screen reader user can't jump there easily; they often end up laboriously cursoring line by line through the extension metadata, screenshots, etc. This makes for a pretty painful experience, to the point where I as a screen reader user tend to dread or even avoid searching for add-ons.

This can be made into a delightful experience as follows:

1. Make all elements with class `Card-header` into heading level 2. You can either use the h2 tag, or if you don't want to mess with tags for styling reasons, use `role="heading" aria-level="2"`.
2. Now that these main headings are h2, the content under "Other users with this extension also installed" needs to have its heading levels adjusted:
    - `.SearchResult-name` should be h3, not h2. Again, you can also use aria-level="3" if you prefer.
    - `.SearchResult-users` should be h4, not h3, if it needs to be a heading at all. Personally, I don't think this is worthy of a heading. I'd suggest changing this to a paragraph, either the `p` tag or `role="paragraph"`.

**Comment by jcsteh** [comment]
This is definitely still a problem. I don't believe there's any additional detail I can provide.
